vars/prepareVMConfig.groovy

```diff
@@ -0,0 +1,38 @@
+def call(vc_hostname, vc_datacenter, vc_cluster, vc_folder, vm_role, vm_hardware_memory_mb, vm_hardware_num_cpus, vm_networks,
+         vc_username, vc_password, vm_template, vm_hdd_size,
+         file_path = 'vm.vars') {
+  def vmConfig = """
+vcenter_hostname: ${vc_hostname}
+vcenter_user: ${vc_username}
+vcenter_password: ${vc_password}
+validate_certs: no
+datacenter_name: ${vc_datacenter}
+cluster_name: ${vc_cluster}
+vmware_folder: ${vc_folder}
+
+vm_template: ${vm_template}
+vm_role: ${vm_role}
+
+vm_hardware:
+  memory_mb: ${vm_hardware_memory_mb}
+  num_cpus: ${vm_hardware_num_cpus}
+
+vm_networks:
+"""
+
+  for(network in vm_networks){
+    network.eachWithIndex{key, value, index -> if(index == 0) vmConfig += "  - "
+                                               else vmConfig += "    "
+                                               vmConfig += "$key: $value\n"}
+  }
+
+  if(vm_hdd_size != null){
+    vmConfig += """
+vm_hdd:
+  - type: thin
+    size: ${vm_hdd_size}
+"""
+  }
+
+  writeFile file: file_path, text: vmConfig
+}
```

Comment:
Note: this is just a suggestion

I'd remove this file. Instead, I'd create a dictionary in groovy, and pass it as `extraVars` parameter of `ansiblePlaybook` step.

Reply:
As far as I know extraVars won't work with nested lists or maps